### PR TITLE
Feature: search in invoices

### PR DESCRIPTION
**Pull request by gmsarates**
I've created the `authorized_payment` method to get information about a authorized payment. 

### Usage: 
One of these parameters are required: `id`, `preapproval_id`, `payment_id` or `payer_id`
```
mercadoPago()->authorized_payment()->search([
    'id' => $id,
    'preapproval_id' => $preapproval_id,
    'payment_id' => $payment_id,
    'payer_id' => $payer_id,
]);
```
By ID
```
mercadoPago()->authorized_payment()->find($id);
```

Official API reference: [click here](https://www.mercadopago.com.br/developers/pt/reference/subscriptions/_authorized_payments_search/get)

### CODE REVIEW

File: src/Resource/AuthorizedPaymentResource.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace WandesCardoso\MercadoPago\Resource;
+
+use Ramsey\Uuid\Uuid;
+use WandesCardoso\MercadoPago\DTO\Payment;
+use WandesCardoso\MercadoPago\DTO\PaymentUpdate;
+
+final class AuthorizedPaymentResource extends MpResource
```
Comment (wandesnet) [inline src/Resource/AuthorizedPaymentResource.php]: To maintain the default, rename it to InvoiceResource

File: src/Traits/MpRequest.php [hunk 2]
```diff
@@ -22,6 +23,11 @@ public function payment(): PaymentResource
         return new PaymentResource($this);
     }
 
+    public function authorized_payment(): AuthorizedPaymentResource
```
Comment (wandesnet) [inline src/Traits/MpRequest.php]: Property and method names must follow the `camelCase` pattern, rename to `Invoice` and change the return type to `InvoiceResource`

**Comment by wandesnet** [comment]
Can you write a test for this new feature?

File: src/Traits/MpRequest.php (hunk 2 above)
Comment (gmsarates) [reply src/Traits/MpRequest.php]: Actually, the resources are using `PascalCase`, in `camelCase` they should call `paymentResource`, `invoiceResource`.  The methods can be using `camelCase` or `snake_case`, as we don't have any one with two names to compare. But there is no problem, I'll follow the pattern!

Comment (wandesnet) [reply src/Traits/MpRequest.php]: Class names must be `PascalCase`, property and method names must follow `camelCase`.

Note: We will only use `camelCase` if necessary, this is not the case in this feature.

**Comment by gmsarates** [comment]
Class names and tests committed

**Comment by wandesnet** [comment]
Thank you for sending the `Pull Request`, an important note, whenever you contribute, remember to create a `new branch`, as it is not correct to make changes directly to the `main branch`.

**Pull request closed without merging**